Mathlib/Topology/LocallyConstant/Algebra.lean

```diff
@@ -291,48 +291,149 @@ theorem coe_algebraMap (r : R) : ⇑(algebraMap R (LocallyConstant X Y) r) = alg
 
 end Algebra
 
+section Eval
+
+/-- Evaluation as a `MonoidHom` -/
+@[to_additive "Evaluation as an `AddMonoidHom`"]
+def evalMonoidHom [MulOneClass Z] (x : X) : LocallyConstant X Z →* Z where
+  toFun f := f x
+  map_mul' _ _ := by simp only [LocallyConstant.coe_mul, Pi.mul_apply]
+  map_one' := rfl
+
+/-- Evaluation as a linear map -/
+def evalₗ [Semiring R] [AddCommMonoid Z]
+  [Module R Z] (x : X) : LocallyConstant X Z →ₗ[R] Z where
+  toFun f := f x
+  map_add' := (evalAddMonoidHom x).map_add'
+  map_smul' _ _ := by simp only [coe_smul, Pi.smul_apply, RingHom.id_apply]
+
+/-- Evaluation as a `RingHom` -/
+def evalRingHom [Semiring Z] (x : X) : LocallyConstant X Z →+* Z where
+  toFun f := f x
+  map_one' := (evalMonoidHom x).map_one'
+  map_mul' := (evalMonoidHom x).map_mul'
+  map_zero' := (evalAddMonoidHom x).map_zero'
+  map_add' := (evalAddMonoidHom x).map_add'
+
+/-- Evaluation as an `AlgHom` -/
+def evalₐ [CommSemiring R] [Semiring Z] [Algebra R Z] (x : X) :
+    LocallyConstant X Z →ₐ[R] Z where
+  toFun f := f x
+  map_one' := (evalMonoidHom x).map_one'
+  map_mul' := (evalMonoidHom x).map_mul'
+  map_zero' := (evalAddMonoidHom x).map_zero'
+  map_add' := (evalAddMonoidHom x).map_add'
+  commutes' r := by simp only [coe_algebraMap, Pi.algebraMap_apply]
```

Comment:
```suggestion
@[simps!]
def evalₐ (R : Type*) [CommSemiring R] [Semiring Z] [Algebra R Z] (x : X) :
    LocallyConstant X Z →ₐ[R] Z where
  toRingHom := evalRingHom x
  commutes' r := by simp
```
Here the `simp` call has to use quite a lot of the lemmas generated by `simps`. But in that case keeping it as `simp` is perfectly fine, and even easier to maintain.

Reply:
Doesn't `rfl` prove this goal?